### PR TITLE
Initial Storybook setup

### PR DESCRIPTION
**Pull request by jojopirker**
These changes enable to use `npm run storybook` to start **storybook** as mentioned in #102. 
There is an example story for the header.
Maybe should be moved to the components folder when doing #99.

What is **not finished**: 
- DataMocking
- fokus only on `components/` rather than `pages/`

What is **finished**:
- TailwindCSS and ChakraUI integration

### CODE REVIEW

File: website/src/stories/Header.stories.jsx
Comment (fozziethebeat) [inline website/src/stories/Header.stories.jsx:4]: style nit, can you delete this newline?

File: website/src/stories/Header.stories.jsx
Comment (fozziethebeat) [inline website/src/stories/Header.stories.jsx:2]: Style nit, can you order these alphabetically?

File: website/src/stories/Header.stories.jsx
Comment (fozziethebeat) [inline website/src/stories/Header.stories.jsx:5]: Is it possible to import from `src/components/Header`? If not, that's okay

**Review by fozziethebeat — APPROVED** [review]

**Comment by fozziethebeat** [comment]
Great first step! I have a few small requested changes then I'll merge this in.

File: website/src/stories/Header.stories.jsx
Comment (jojopirker) [reply website/src/stories/Header.stories.jsx]: moved them into the same folder, think thats inline with #99

**Comment by jojopirker** [comment]
Deleted the empty lines :)

I also moved the files so its more inline with future structure.

**Comment by AbdBarho** [comment]
would make sense the have the stories in another folder? i.e. not between all the components? maybe in `src/stories/...`

I am suggesting this because stories are more of "write and forget", otherwise the files would be in the way all the time.

@jojopirker @fozziethebeat  what do you think?

**Comment by fozziethebeat** [comment]
> would make sense the have the stories in another folder? i.e. not between all the components? maybe in `src/stories/...`
> 
> I am suggesting this because stories are more of "write and forget", otherwise the files would be in the way all the time.
> 
> @jojopirker @fozziethebeat what do you think?

This is definitely a matter of preference and probably past experiences.  

I like how RedwoodJS put the test and stories in the same directory as the component so that when you make changes to the component you're reminded to check the story or test and don't have to go find the corresponding file in some other parallel directory structure.  Let's try this for now and then re-evaluate.

File: website/src/pages/index.tsx
Comment (fozziethebeat) [inline website/src/pages/index.tsx:10]: This looks wrong, I think you meant to change line 9

File: website/src/components/Header/Header.stories.jsx
Comment (fozziethebeat) [inline website/src/components/Header/Header.stories.jsx:10]: we can probably delete this in the code.  If people get confused we can add explanations in a README.

File: website/src/components/Header/Header.tsx
Comment (fozziethebeat) [inline website/src/components/Header/Header.tsx:9]: Can these be absolute imports (`src/components/Avatar/Avatar`)?

**Review by fozziethebeat — APPROVED** [review]
Looks good after a few tiny changes and running prettier.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
Holding off on the approval until the merge fixes are in.  I'd like to make sure those are all addressed and no new issues get introduced.

**Comment by jojopirker** [comment]
Should include all the fixes now.

File: website/src/pages/create/summarize_story.tsx
Comment (AbdBarho) [inline website/src/pages/create/summarize_story.tsx:8]: can we replace `@` with `src`?
I know vscode automatically added it....

File: website/src/pages/evaluate/rate_summary.tsx
Comment (AbdBarho) [inline website/src/pages/evaluate/rate_summary.tsx:10]: same here

**Review by AbdBarho — COMMENTED** [review]
Thanks!
Could you please also resolve the merge conflicts so we can merge this?

File: website/src/pages/create/summarize_story.tsx
Comment (jojopirker) [reply website/src/pages/create/summarize_story.tsx]: I removed the line entirely while merging..

File: website/src/pages/evaluate/rate_summary.tsx
Comment (jojopirker) [reply website/src/pages/evaluate/rate_summary.tsx]: also gone :D

File: website/.storybook/main.js
Comment (AbdBarho) [inline website/.storybook/main.js:20]: why is this necessary?

File: website/.storybook/main.js
Comment (AbdBarho) [inline website/.storybook/main.js:26]: why is this necessary?

File: website/.storybook/preview.js
Comment (AbdBarho) [inline website/.storybook/preview.js:1]: this should be replaced in the future with a theme provider there includes chakra ui & tailwind stuff.

File: website/.storybook/preview.js
Comment (AbdBarho) [inline website/.storybook/preview.js:20]: oof...

what's the problem here exactly?

**Review by AbdBarho — COMMENTED** [review]
while installing  locally, I get the following in the console:
```sh
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @mdx-js/react@1.6.22
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"18.2.0" from the root project
npm WARN   128 more (@chakra-ui/accordion, @chakra-ui/alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm WARN node_modules/@mdx-js/react
npm WARN   @mdx-js/react@"^1.6.22" from @storybook/addon-docs@6.5.15
npm WARN   node_modules/@storybook/addon-docs
npm WARN
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm WARN   node_modules/@mdx-js/react
npm WARN     @mdx-js/react@"^1.6.22" from @storybook/addon-docs@6.5.15
npm WARN     node_modules/@storybook/addon-docs
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-element-to-jsx-string@14.3.4
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"18.2.0" from the root project
npm WARN   128 more (@chakra-ui/accordion, @chakra-ui/alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN node_modules/react-element-to-jsx-string
npm WARN   react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.15
npm WARN   node_modules/@storybook/react
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN   node_modules/react-element-to-jsx-string
npm WARN     react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.15
npm WARN     node_modules/@storybook/react
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-element-to-jsx-string@14.3.4
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   react-dom@"18.2.0" from the root project
npm WARN   40 more (@chakra-ui/modal, @chakra-ui/portal, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN node_modules/react-element-to-jsx-string
npm WARN   react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.15
npm WARN   node_modules/@storybook/react
npm WARN
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN   node_modules/react-element-to-jsx-string
npm WARN     react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.15
npm WARN     node_modules/@storybook/react
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-inspector@5.1.1
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"18.2.0" from the root project
npm WARN   128 more (@chakra-ui/accordion, @chakra-ui/alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN node_modules/react-inspector
npm WARN   react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.15
npm WARN   node_modules/@storybook/addon-actions
npm WARN
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN   node_modules/react-inspector
npm WARN     react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.15
npm WARN     node_modules/@storybook/addon-actions
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
```
Is this expected? storybook already has react 18 support, not sure why it is complaining  here.

File: website/.storybook/main.js
Comment (jojopirker) [reply website/.storybook/main.js:26]: necessary for for chakra to work properly, see [here](https://chakra-ui.com/getting-started/with-storybook)

File: website/.storybook/preview.js
Comment (jojopirker) [reply website/.storybook/preview.js:20]: little hacky solution because next handles images in a way, that simply adding the public folder to storybook would not work... i used the solution from [here](https://dev.to/jonasmerlin/how-to-use-the-next-js-image-component-in-storybook-1415)

Comment (AbdBarho) [reply website/.storybook/preview.js:20]: Damn, very nasty, could you  please document this in the code? so that future developers can understand why.

File: website/.storybook/main.js
Comment (jojopirker) [reply website/.storybook/main.js:20]: ` typescript: { reactDocgen: false },` was also necessary for absolute imports... i just cant find the source where i found the solution

Comment (jojopirker) [reply website/.storybook/main.js:20]: found [it](https://github.com/storybookjs/storybook/issues/15336#issuecomment-888528747), had an error when trying to use the absolute imports, that was a fix for it..

Comment (AbdBarho) [reply website/.storybook/main.js:20]: https://github.com/storybookjs/storybook/issues/15336#issuecomment-888528747

Comment (AbdBarho) [reply website/.storybook/main.js:20]: okay, so the reason is absolute imports, maybe you could document that.

In the link above some users reported added webpack v5 to dev dependencies and everything just worked, however, it pains me that we have to do this.

The offical package has not been updated in a while, which is a bad sign: https://www.npmjs.com/package/@storybook/react-docgen-typescript-plugin


We lose react-docgen, but I think that is ok.

For the webpack final, can you please make it explicit that it is only resolving aliases, and not replacing modules.

```js

    config.resolve.alias = {
      ...config.resolve.alias,
      src: path.resolve(__dirname, "../src"),
    };
    return config;
```

**Review by fozziethebeat — APPROVED** [review]
This looks pretty good.  Please make sure the merge with header.tsx works smoothly and nothing gets dropped.

**Comment by jojopirker** [comment]
merged the previous changes... i also moved all the files of the header into `/components/header` so they are all in one place, and added and index file in the folder for better import

**Comment by AbdBarho** [comment]
> merged the previous changes... i also moved all the files of the header into `/components/header` so they are all in one place, and added and index file in the folder for better import

Thanks! I see a `docker-compose 2.yaml` file, this was not intended right?

**Comment by jojopirker** [comment]
yup... i still need to learn how to properly merge with git^^

**Review by AbdBarho — APPROVED** [review]

**Pull request merged**
